pkg/network/ebpf/c/prebuilt/tracer.c

```diff
@@ -423,6 +426,42 @@ int kprobe__tcp_set_state(struct pt_regs* ctx) {
     return 0;
 }
 
+SEC("kprobe/tcp_connect")
+int kprobe__tcp_connect(struct pt_regs *ctx) {
+    u64 pid_tgid = bpf_get_current_pid_tgid();
+    log_debug("kprobe/tcp_connect: tgid: %u, pid: %u\n", pid_tgid >> 32, pid_tgid & 0xFFFFFFFF);
+    struct sock *skp = (struct sock *)PT_REGS_PARM1(ctx);
+
+    bpf_map_update_elem(&tcp_connect_sock_pid, &skp, &pid_tgid, BPF_ANY);
```

Comment:
Does the `pid_fd_by_sock` map already do what you need here?

Reply:
It does, for the purpose of this PR. However, that new map should allow us to detect failing connections on the client-side later on, by looking at whether the `sock` is in the map when `tcp_close` is called.

Reply:
Yes, if you're using for that purpose, the name should more explicit.